File: finalyearproject/repos/dkd-b2c-order/src/http/health.ts
import type { ServerResponse } from "node:http";
import { CONTRACT_VERSION, GENERATOR_VERSION } from "@dokandar/platform-sdk";

export function json(res: ServerResponse, status: number, body: unknown): void {
  res.writeHead(status, { "Content-Type": "application/json" });
  res.end(JSON.stringify(body));
}

export function health(res: ServerResponse): void {
  json(res, 200, { success: true, data: { status: "ok" } });
}
export function live(res: ServerResponse): void {
  json(res, 200, { success: true, data: { status: "alive" } });
}
export function ready(res: ServerResponse, isReady: boolean): void {
  if (!isReady) { json(res, 503, { success: false, data: { status: "not-ready" } }); return; }
  json(res, 200, { success: true, data: { status: "ready" } });
}
export function version(res: ServerResponse): void {
  json(res, 200, { success: true, data: { contractVersion: CONTRACT_VERSION, sdkGenerator: GENERATOR_VERSION } });
}

File: finalyearproject/repos/dkd-logistics/internal/httpx/health.go
package httpx

import (
	"encoding/json"
	"net/http"

	"gitlab.com/final-year-project3354127/logistics-svc/internal/buildinfo"
	dkdplatform "gitlab.com/final-year-project3354127/dkd-platform-libs/sdk/go"
)

func writeJSON(w http.ResponseWriter, status int, body any) {
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(status)
	_ = json.NewEncoder(w).Encode(body)
}

func health(w http.ResponseWriter, _ *http.Request) {
	writeJSON(w, http.StatusOK, map[string]any{"success": true, "data": map[string]string{"status": "ok"}})
}

func live(w http.ResponseWriter, _ *http.Request) {
	writeJSON(w, http.StatusOK, map[string]any{"success": true, "data": map[string]string{"status": "alive"}})
}

func readyHandler(ready func() bool) http.HandlerFunc {
	return func(w http.ResponseWriter, _ *http.Request) {
		if ready != nil && !ready() {
			writeJSON(w, http.StatusServiceUnavailable, map[string]any{"success": false, "data": map[string]string{"status": "not-ready"}})
			return
		}
		writeJSON(w, http.StatusOK, map[string]any{"success": true, "data": map[string]string{"status": "ready"}})
	}
}

func version(w http.ResponseWriter, _ *http.Request) {
	writeJSON(w, http.StatusOK, map[string]any{"success": true, "data": map[string]string{
		"version":         buildinfo.Version,
		"gitSha":          buildinfo.GitSha,
		"buildTime":       buildinfo.BuildTime,
		"contractVersion": dkdplatform.ContractVersion,
		"sdkGenerator":    dkdplatform.GeneratorVersion,
	}})
}
